addons/common/fnc_compatibleMagazines.sqf

```diff
@@ -22,56 +23,42 @@ Author:
 ---------------------------------------------------------------------------- */
 SCRIPT(compatibleMagazines);
 
-params [["_weapon", "", ["", configNull]]];
+params [["_weapon", "", ["", configNull]], ["_allMuzzles", false, [false]]];
 
 if (_weapon isEqualType "") then {
     _weapon = configFile >> "CfgWeapons" >> _weapon;
 };
 
-private _cacheKey = str _weapon;
+private _cacheKey = format ["%1#%2",_weapon,_allMuzzles];
 
```

Comment:
```suggestion

if (_cacheKey == "") exitWith {
    ERROR_1("Weapon Does Not Exist %1",_this);
    []
};
```

Reply:
cacheKey will never be "" because `format ["%1#%2",_weapon,_allMuzzles];`
this was only there for the getVar "" thing, not really needed